packages/web3-eth/src/errors.ts

```diff
@@ -1,21 +1,159 @@
+	public constructor() {
+		// TODO - Discuss what to pass as value, would it make sense to pass common, chain, and hardfork for error message here?
+		super(
+			'N/A',
+			'Please provide the @ethereumjs/common object or the chain and hardfork property but not all together.',
+		);
+	}
+}
+
+export class MissingChainOrHardforkError extends Web3Error {
+	public constructor(value: { chain: string | undefined; hardfork: string | undefined }) {
+		// TODO - Discuss what to pass as value
+		super(
+			'N/A',
+			`When specifying chain and hardfork, both values must be defined. Received "chain": ${
+				value.chain ?? 'undefined'
+			}, "hardfork": ${value.hardfork ?? 'undefined'}`,
+		);
+	}
+}
+
+export class MissingGasError extends Web3Error {
+	public constructor(value: {
+		gas: Numbers | undefined;
+		gasLimit: Numbers | undefined;
+		maxPriorityFeePerGas: Numbers | undefined;
+		maxFeePerGas: Numbers | undefined;
+	}) {
+		super(
+			`gas: ${value.gas ?? 'undefined'}, gasLimit: ${
+				value.gasLimit ?? 'undefined'
+			}, maxPriorityFeePerGas: ${value.maxPriorityFeePerGas ?? 'undefined'}, maxFeePerGas: ${
+				value.maxFeePerGas ?? 'undefined'
+			}`,
+			'"gas" is missing',
+		);
+	}
+}
+
+export class InvalidGasOrGasPrice extends Web3Error {
+	public constructor(value: { gas: Numbers | undefined; gasPrice: Numbers | undefined }) {
+		super(
+			`gas: ${value.gas ?? 'undefined'}, gasPrice: ${value.gasPrice ?? 'undefined'}`,
+			'Gas or gasPrice is lower than 0',
+		);
+	}
+}
+
+export class InvalidMaxPriorityFeePerGasOrMaxFeePerGas extends Web3Error {
+	public constructor(value: {
+		maxPriorityFeePerGas: Numbers | undefined;
+		maxFeePerGas: Numbers | undefined;
+	}) {
+		super(
+			`maxPriorityFeePerGas: ${value.maxPriorityFeePerGas ?? 'undefined'}, maxFeePerGas: ${
+				value.maxFeePerGas ?? 'undefined'
+			}`,
+			'maxPriorityFeePerGas or maxFeePerGas is lower than 0',
+		);
+	}
+}
+
+export class Eip1559GasPriceError extends Web3Error {
+	public constructor(value: unknown) {
+		super(value, "eip-1559 transactions don't support gasPrice");
+	}
+}
+
+export class UnsupportedFeeMarketError extends Web3Error {
+	public constructor(value: {
+		maxPriorityFeePerGas: Numbers | undefined;
+		maxFeePerGas: Numbers | undefined;
+	}) {
+		super(
+			`maxPriorityFeePerGas: ${value.maxPriorityFeePerGas ?? 'undefined'}, maxFeePerGas: ${
+				value.maxFeePerGas ?? 'undefined'
+			}`,
+			"pre-eip-1559 transaction don't support maxFeePerGas/maxPriorityFeePerGas",
+		);
+	}
+}
+
+export class InvalidTransactionObjectError extends Web3Error {
+	public constructor(value: unknown) {
+		super(value, 'invalid transaction object');
+	}
+}
+
+export class InvalidNonceOrChainIdError extends Web3Error {
+	public constructor(value: { nonce: Numbers | undefined; chainId: Numbers | undefined }) {
+		super(
+			`nonce: ${value.nonce ?? 'undefined'}, chainId: ${value.chainId ?? 'undefined'}`,
+			'Nonce or chainId is lower than 0',
+		);
+	}
+}
```

Comment:
In all above cases of different errors, I would suggest to assign different codes to each error as well. e.g. https://github.com/ChainSafe/web3.js/blob/5a00ab5b4098654280811bf098030e7db485441c/packages/web3-common/src/constants.ts#L24-L32